cli/cmdlineparser.cpp

```diff
@@ -592,10 +592,10 @@ bool CmdLineParser::parseFromArgs(int argc, const char* const argv[])
 
             // Print help
             else if (std::strcmp(argv[i], "-h") == 0 || std::strcmp(argv[i], "--help") == 0) {
-                mPathNames.clear();
-                mShowHelp = true;
+                // TODO: make this an exclusive option
```

Comment:
"exclusive" option means that it should be processed separately so e.g. `--library=missing-lib --help` would not bailout with a library error but simply shows the help text. So we could scan for those before before handling all the others.

I tried a few different commands and the behavior is not consistent. Some just do the "left-to-right" until they hit the bail out options and others treat them as truly exclusive. I would like to go with the latter but that will be done in an upcoming PR.